catalog/rocks/validate.go

```diff
@@ -20,35 +20,63 @@ var (
 )
 
 var (
-	ErrInvalidValue = errors.New("invalid value")
-	ErrInvalidType  = errors.New("invalid type")
+	ErrInvalidType   = errors.New("invalid type")
+	ErrInvalidValue  = errors.New("invalid value")
+	ErrRequiredValue = errors.New("required value")
 )
 
 type ValidateFunc func(v interface{}) error
 
-func Validate(name string, value interface{}, fn ValidateFunc) error {
-	err := fn(value)
-	if err != nil {
-		return fmt.Errorf("%s: %w", name, err)
+type ValidateArg struct {
+	Name  string
+	Value interface{}
+	Fn    ValidateFunc
+}
+
+func Validate(args []ValidateArg) error {
+	for _, arg := range args {
+		err := arg.Fn(arg.Value)
+		if err != nil {
+			return fmt.Errorf("argument %s: %w", arg.Name, err)
+		}
 	}
 	return nil
 }
 
-func ValidateNonEmptyString(v interface{}) error {
-	s, ok := v.(string)
+func MakeValidateOptional(fn ValidateFunc) ValidateFunc {
+	return func(v interface{}) error {
```

Comment:
nil `v` isn't an option?

Reply:
for string or stringer no - but I'll add support